nesta/core/orms/orm_utils.py

```diff
@@ -630,17 +630,23 @@ def insert_data(db_env, section, database, Base,
                                data=data,
                                low_memory=low_memory)
     objs, existing_objs, failed_objs = response
-    # Open a transaction
-    session = get_session(db_env, section, database, Base)
-    if merge_non_null:
-        session.execute(existing_objs)  # Drop existing objs if merging
+
+    # Prepare for transactions
+    engine = get_mysql_engine(db_env, section, database)
+
+    # Drop existing objs if merging
+    with db_session(engine) as session:
+        try_until_allowed(Base.metadata.create_all, 
+                          session.get_bind())
+        if merge_non_null:
+            session.execute(existing_objs)
+
     # Insert data in chunks
     for chunk in split_batches(objs, insert_chunksize):
-        stmt = insert(_class).values(chunk)
-        session.execute(stmt)
-    # Commit and close
-    session.commit()
-    session.close()
+        with db_session(engine) as session:
```

Comment:
if the total length of time of the transaction is too long (as is the case when inserting several million 700-dim vectors) then a single session will die eventually. Using smaller transactions is the only way forward, and doesn't affect other pipelines, and is anyway better practice.